### PR TITLE
Updated install instructions and readme to be more clear

### PR DESCRIPTION
**Pull request by Innixma**
*Issue #, if available:*
#722 

*Description of changes:*

- Updated install instructions and readme to be more clear about differences in v0.0.14 and mainline.
- Made it clearer that AutoGluon is not limited to only deep learning.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by szha** [comment]
Job PR-727-2 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-727/2/index.html

File: README.md [hunk 1]
```diff
@@ -4,13 +4,13 @@
   <img src="https://user-images.githubusercontent.com/16392542/77208906-224aa500-6aba-11ea-96bd-e81806074030.png" width="350">
 </div>
 
-## AutoML Toolkit for Deep Learning
+## AutoML Toolkit for Tabular, Text, and Vision
```
Comment (jwmueller) [inline README.md:7]: I'd change to: "for Text, Image, and Tabular Data" to be grammatically correct.

File: docs/install-warning.rst [hunk 2]
```diff
@@ -0,0 +1,4 @@
+.. warning::
+    This documentation is based off of the unreleased bleeding-edge mainline branch which may have frequent API breaking changes.
```
Comment (jwmueller) [inline docs/install-warning.rst:2]: "unreleased bleeding-edge" -> "pre-release"

**Review by jwmueller — APPROVED** [review]
just have minor text improvements

File: README.md (hunk 1 above)
Comment (Innixma) [reply README.md]: added

File: docs/install-warning.rst (hunk 2 above)
Comment (Innixma) [reply docs/install-warning.rst]: added

**Comment by szha** [comment]
Job PR-727-3 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-727/3/index.html

**Pull request merged**